pkg/recipes/terraform/config/config.go

```diff
@@ -96,24 +97,32 @@ func (cfg *TerraformConfig) Save(ctx context.Context, workingDir string) error {
 	// JSON configuration syntax for Terraform requires the file to be named with .tf.json suffix.
 	// https://developer.hashicorp.com/terraform/language/syntax/json
 
-	// Convert the Terraform config to JSON
-	jsonData, err := json.MarshalIndent(cfg, "", "  ")
-	if err != nil {
+	// Create a buffer to write the JSON to
+	buf := &bytes.Buffer{}
+	enc := json.NewEncoder(buf)
+	enc.SetEscapeHTML(false)
+	enc.SetIndent("", "  ") // Indent with 2 spaces to make the JSON file human-readable and consistent with codebase.
+
+	// Encode the Terraform config to JSON. JSON encoding is being used to ensure that special characters
+	// in the original text are preserved when writing to the file.
+	if err := enc.Encode(cfg); err != nil {
```

Comment:
Can you please elaborate the comment with specific example on what breaks if we don't attach the buffer to the encoder first? This will help anyone reading the code easily get the context on which fields require this extra step.

Reply:
updated comments with a detailed example. Pls check.

Reply:
Thanks